contract/src/wallet/WalletV5R1Contract.kt

```diff
@@ -0,0 +1,253 @@
+                cellBuilder.storeBytes(value.publicKey.key.toByteArray())
+                cellBuilder.storeBoolean(false)
+            }
+        }
+    }
+
+    public companion object {
+        public val CODE: Cell by lazy(LazyThreadSafetyMode.PUBLICATION) {
+            BagOfCells(
+                Base64.decode("te6cckECFAEAAoEAART/APSkE/S88sgLAQIBIAINAgFIAwQC3NAg10nBIJFbj2Mg1wsfIIIQZXh0br0hghBzaW50vbCSXwPgghBleHRuuo60gCDXIQHQdNch+kAw+kT4KPpEMFi9kVvg7UTQgQFB1yH0BYMH9A5voTGRMOGAQNchcH/bPOAxINdJgQKAuZEw4HDiEA8CASAFDAIBIAYJAgFuBwgAGa3OdqJoQCDrkOuF/8AAGa8d9qJoQBDrkOuFj8ACAUgKCwAXsyX7UTQcdch1wsfgABGyYvtRNDXCgCAAGb5fD2omhAgKDrkPoCwBAvIOAR4g1wsfghBzaWduuvLgin8PAeaO8O2i7fshgwjXIgKDCNcjIIAg1yHTH9Mf0x/tRNDSANMfINMf0//XCgAK+QFAzPkQmiiUXwrbMeHywIffArNQB7Dy0IRRJbry4IVQNrry4Ib4I7vy0IgikvgA3gGkf8jKAMsfAc8Wye1UIJL4D95w2zzYEAP27aLt+wL0BCFukmwhjkwCIdc5MHCUIccAs44tAdcoIHYeQ2wg10nACPLgkyDXSsAC8uCTINcdBscSwgBSMLDy0InXTNc5MAGk6GwShAe78uCT10rAAPLgk+1V4tIAAcAAkVvg69csCBQgkXCWAdcsCBwS4lIQseMPINdKERITAJYB+kAB+kT4KPpEMFi68uCR7UTQgQFB1xj0BQSdf8jKAEAEgwf0U/Lgi44UA4MH9Fvy4Iwi1woAIW4Bs7Dy0JDiyFADzxYS9ADJ7VQAcjDXLAgkji0h8uCS0gDtRNDSAFETuvLQj1RQMJExnAGBAUDXIdcKAPLgjuLIygBYzxbJ7VST8sCN4gAQk1vbMeHXTNC01sNe")
+            ).first()
+        }
+
+        public const val OP_SEND: Int = 0
+
+        public fun address(privateKey: PrivateKeyEd25519, walletId: WalletId): AddrStd {
+            val stateInitRef = stateInit(Data(0, privateKey.publicKey(), walletId)) // Initial sequence number is 0
+            val hash = stateInitRef.hash()
+            return AddrStd(walletId.workchain, hash)
+        }
+
+        public fun stateInit(
+            data: Data,
+        ): CellRef<StateInit> {
+            val dataCell = buildCell {
+                storeTlb(Data, data)
+            }
+            return CellRef(
+                StateInit(CODE, dataCell),
+                StateInit
+            )
+        }
+
+        public fun transferMessage(
+            address: MsgAddressInt,
+            stateInit: StateInit?,
+            privateKey: PrivateKeyEd25519,
+            validUntil: Int,
+            seqno: Int,
+            walletId: WalletId,
+            vararg transfers: WalletTransfer
+        ): Message<Cell> {
+            val info = ExtInMsgInfo(
+                src = AddrNone,
+                dest = address,
+                importFee = Coins()
+            )
+
+            val maybeStateInit =
+                Maybe.of(stateInit?.let {
+                    Either.of<StateInit, CellRef<StateInit>>(
+                        null,
+                        CellRef(value = it, StateInit)
+                    )
+                })
+
+
+            val transferBody = createTransferMessageBody(
+                privateKey,
+                validUntil,
+                seqno,
+                walletId,
+                *transfers
+            )
+
+            val body = Either.of<Cell, CellRef<Cell>>(null, CellRef(value = transferBody, AnyTlbConstructor))
```

Comment:
Permanently set to cell? There should be a way to change this, there is a MessageLayout class for this

Reply:
I have referenced the above two codes. 
https://github.com/ton-blockchain/ton-kotlin/blob/main/contract/src/wallet/WalletV3Contract.kt
https://github.com/ton-blockchain/ton-kotlin/blob/main/contract/src/wallet/WalletV4R2Contract.kt

Do you wanna like this?  https://github.com/ton-blockchain/ton-kotlin/pull/131/commits/cc40397539f327123499794d72c63ade73f83f86